src/config.rs

```diff
@@ -310,30 +311,48 @@ impl<'de> Deserialize<'de> for Config {
             return Ok(Config::from_legacy(raw));
         }
 
+        use serde::de::Error;
         let mut table = match raw {
             Value::Table(t) => t,
             _ => {
-                use serde::de::Error;
                 return Err(D::Error::custom(
                     "A config file should always be a toml table",
                 ));
             }
         };
 
-        let book: BookConfig = table
-            .remove("book")
-            .and_then(|value| value.try_into().ok())
-            .unwrap_or_default();
+        let book = if let Some(book) = table.remove("book") {
+            match book.try_into() {
+                Ok(b) => b,
+                Err(e) => {
+                    return Err(D::Error::custom(e));
+                }
+            }
+        } else {
+            BookConfig::default()
+        };
```

Comment:
I think this can be kept to something closer to original like this:

```suggestion
        let book: BookConfig = table
            .remove("book")
            .map(|book| book.try_into().map_err(D::Error::custom))
            .transpose()?
            .unwrap_or_default();
```